preferences-rest/src/main/java/no/sikt/nva/person/licenceinfo/rest/UpsertLicenseInfoHandler.java

```diff
@@ -0,0 +1,57 @@
+package no.sikt.nva.person.licenceinfo.rest;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import no.sikt.nva.person.preferences.commons.model.LicenseInfoDto;
+import no.sikt.nva.person.preferences.commons.model.LicenseInfoDao;
+import no.sikt.nva.person.preferences.commons.service.IndexService;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+import nva.commons.core.JacocoGenerated;
+
+import java.net.HttpURLConnection;
+
+import static no.sikt.nva.person.Constants.TABLE_NAME;
+import static no.sikt.nva.person.Constants.dontMatchCustomerAndPerson;
+
+public class UpsertLicenseInfoHandler extends ApiGatewayHandler<LicenseInfoDto, LicenseInfoDto> {
+
+    private final IndexService<LicenseInfoDao> dynamoDbService;
+
+    @JacocoGenerated
+    public UpsertLicenseInfoHandler() {
+        this(new IndexService<>( TABLE_NAME, LicenseInfoDao.class));
+    }
+
+    public UpsertLicenseInfoHandler(IndexService<LicenseInfoDao> personPreferencesService) {
+        super(LicenseInfoDto.class);
+        this.dynamoDbService = personPreferencesService;
+    }
+
+
+    @Override
+    protected void validateRequest(LicenseInfoDto licenseInfoDto, RequestInfo requestInfo, Context context)
+            throws ApiGatewayException {
+        if (dontMatchCustomerAndPerson(requestInfo)) {
+            throw new UnauthorizedException();
+        }
+    }
+
+
+    @Override
+    protected LicenseInfoDto processInput(LicenseInfoDto input, RequestInfo requestInfo, Context context)
+            throws ApiGatewayException {
+        return LicenseInfoDao.builder()
+                .withId(requestInfo.getPersonCristinId())
+                .licenseUri(input.licenseUri())
+                .build()
+                .upsert(dynamoDbService)
+                .toDto();
+    }
```

Comment:
Har aksesserer vi Dao direkte fra handler + dao persisterer seg selv. Hadde det kanskje vært bedre å innføre Business object som LicenseInfo?

Reply:
i DDD er det dao som persisterer ved hjelp av en generell tjeneste.
og tjenesten skal levere dao tilbake igjen.
Derfor må den transformeres til DTO før vi sender den av gårde.

Reply:
Ja, Dao som skal persisteres, men ikke gjøre noe annet enn det? For eksempel LicenseInfo.upsert(service) og inn i upsert metoden kjører vi konverteringen til Dao.

Reply:
Bakgrunn her var å bruke active record som skal kun lagre seg selv (som den også gjør nå), slik at vi ikke trenger en service bortsett fra repository. Du får bestemme hva du liker mest :)